forum/qa-plugin/report-reason/q2apro-flag-reasons-page.php

```diff
@@ -0,0 +1,209 @@
+<?php
+
+class q2apro_flag_reasons_page extends q2apro_flag_reasons_validation {
+
+    private $directory;
+    private $urlToRoot;
+
+    protected $CUSTOM_REPORT_REASON_ID = 0;
```

Comment:
wielkimi literami raczej zapisujemy stałe, nie spotkałem się z powszechnym używaniem wielkich do właściwości

Reply:
Początkowo chciałem zrobić z tego stałą, ale napotkałem na problemy ze składnią PHP, więc zostawiłem jako property. To jest docelowo ustawiane w kostruktorze tylko raz - a w innych miejscach jest jedynie odczyt - stąd pomysł aby to była stała (nie udało się przypisać do niej wartości zwróconej przez funkcję, więc docelowe przypisanie przeniosłem właśnie do konstruktora).
Czy w takim razie lepiej jest zapisać to jako zwykłe property (z małych liter), czy można zrobić z tego stałą i inicjować ją w miejscu utworzenia?

Reply:
Jak dla mnie sensowniej zrobić z tego zwykłą właściwość i tyle

Reply:
Jakie to były problemy ze składnią? 
albo robisz protected const, albo protected static, a później:
self:: lub static:: wewnątrz tej klasy.